refactor: rename route to polyline

KhaledSakr opened the pull request:
BREAKING CHANGE: this completely removes `Route` component and replaces it with `Polyline`. The `arrow` prop is also completely removed from the component. For more information on direction arrows, check the readme.

Additionally, `lng` is now used consistently instead of `lon` to stay in line with the HERE SDK conventions.

## What the PR Includes
- [x] Rename `Route` component to `Polyline`.
- [x] Rename `lon` to `lng`.
- [x] Remove `arrow` prop.

## Checklist
- [ ] Tests are added.
- [ ] Manual testing instructions are added.
- [ ] Configs are added/adapted if applicable.
- [ ] Issue is properly linked if applicable.

jlabeit reviewed (commented):
Do you have some more details on "somehow get the package to work in the CargoApps"? 

When I try to use `yarn link` the map directly crashes with:
```
react.development.js?e669:1630 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'useRef')
```
in the `HEREMap` file: 

```
    var uniqueIdRef = (0, react_1.useRef)((0, lodash_1.uniqueId)());
```

The only think I've adapted is [upgrading `node-sass` because](https://github.com/impargo/react-here-maps/pull/13) I am using node 18 locally.

KhaledSakr commented:
I think the easiest way is to release a prepublish version:
yarn transpile && yarn publish --no-git-tag-version --new-version 1.8.0-pre-1

KhaledSakr commented:
Here I released a new version for you: https://github.com/impargo/react-here-maps/pkgs/npm/react-here-maps/39775758
```
yarn add @impargo/react-here-maps@1.8.0-pre-1
```

jlabeit commented on src/HEREMap.tsx:
```diff
@@ -43,150 +41,186 @@ export interface HEREMapState {
-  public getMap: () => H.Map;
-  public setCenter: (point: H.geo.IPoint) => void;
-  public setZoom: (zoom: number) => void;
+  const [map, setMap] = useState(null);
```
```suggestion
  const [map, setMap] = useState<H.Map>(null);
```

jlabeit commented on src/HEREMap.tsx:
```diff
@@ -43,150 +41,186 @@ export interface HEREMapState {
+  const unmountedRef = useRef(false);
+
+  const defaultLayersRef = useRef(null);
+  const trafficOverlayLayerRef = useRef(null);
+  const truckOverlayLayerRef = useRef(null);
+  const truckCongestionLayerRef = useRef(null);
```
```suggestion
  const trafficOverlayLayerRef = useRef<H.map.layer.TileLayer>(null);
  const truckOverlayLayerRef = useRef<H.map.layer.TileLayer>(null);
  const truckCongestionLayerRef = useRef<H.map.layer.TileLayer>(null);
```

jlabeit commented on src/HEREMap.tsx:
```diff
@@ -43,150 +41,186 @@ export interface HEREMapState {
+  };
+
+  useEffect(() => {
+    return () => {
+      unmountedRef.current = true;
+    };
```
Minor: I am wondering if it would make sense to move this logic into the below `useEffect` hook to the `map?.dispose()` as `unmountedRef` only is set to be used in this hook and belongs to the concern "loading and initializing scripts".

KhaledSakr replied:
I agree, that makes sense 👍🏼

jlabeit reviewed (commented):
Just a few minor comments otherwise looks and works good to me :+1:

KhaledSakr committed:
refactor: rename route to polyline

BREAKING CHANGE: this completely removes `Route` component and replaces it
with `Polyline`. The `arrow` prop is also completely removed from the
component. For more information on direction arrows, check the readme.

Additionally, `lng` is now used consistently instead of `lon` to stay in
line with the HERE SDK conventions.

KhaledSakr commented:
/publish

david-paulus-saad commented:
RUN URL: https://github.com/impargo/react-here-maps/actions/runs/12812746356
PACKAGE VERSION: 3.0.0-pre-24
